logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/session/RegisterTokenUseCase.kt

```diff
@@ -0,0 +1,13 @@
+package com.wire.kalium.logic.feature.session
+
+import com.wire.kalium.logic.data.event.EventRepository
+import com.wire.kalium.logic.functional.suspending
+import com.wire.kalium.network.api.notification.pushToken.PushTokenRequestBody
+
+class RegisterTokenUseCase(
+    private val eventRepository: EventRepository
+) {
+    suspend operator fun invoke(body: PushTokenRequestBody) = suspending {
```

Comment:
This UseCase exposes `Either` type to the app which is fine for Android/JVM but may cause issues on iSO so as a team we decided not expose Either outside kalium. You can create a `sealed class` to represent the error and the success state.